### PR TITLE
Hayoo: Updated to be restored (fixes #1399)

### LINKED ISSUE #1399: Find replacement for Hayoo API
@MrChrisW mentioned in https://github.com/duckduckgo/zeroclickinfo-spice/pull/1376#issuecomment-68138340 that the endpoint no longer works. A replacement would be nice.

https://duck.co/ia/view/hayoo

### PR DESCRIPTION
**Pull request by killerfish**
fixes #1399 
@jagtalon This seemed like the best way to organize the info, let me know if this needs changes.
SS below
hayoo

### CODE REVIEW

File: lib/DDG/Spice/Hayoo.pm [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use DDG::Spice;
+
+primary_example_queries "hayoo Prelude map";
+description "Search Haskell APIs";
+name "Hayoo";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Hayoo.pm";
+icon_url "/i/hackage.haskell.org.ico";
+topics "programming", "sysadmin";
+category "programming";
+attribution github => ['https://github.com/headprogrammingczar','headprogrammingczar'];
```
Comment (moollaza) [inline lib/DDG/Spice/Hayoo.pm]: Feel free to add yourself @killerfish

**Comment by moollaza** [comment]
It seems like the whole point of Hayoo is to be able to search by function, because Hackage already lets you search by Package name (i.e. Prelude).

Maybe we should file a bug/email the Hayoo maintainers so `Prelude.map` will work again?

This is looking good so far though @killerfish

**Comment by moollaza** [comment]
Actually I may be wrong, Hayoo provides example queries: http://hayoo.fh-wedel.de/examples so perhaps this is good enough.

**Comment by jagtalon** [comment]
@killerfish looks good! :+1: Make a PR for your attribution!

**Comment by moollaza** [comment]
@killerfish congrats! It's live: https://duckduckgo.com/?q=hayoo+Prelude+map&ia=software

**Pull request merged**